### PR TITLE
Wildcard ports

### LINKED ISSUE #59: services should not require pods and pod-templates to declare explicit hostPort's
since merging #54 services can now connect to pods but only if those pods (or pod templates) are declared with a hostPort != 0. not including a hostPort definition is the same as assigning a value of zero, which triggers the default k8s behavior of not publishing the containerPort to the host at all. specifying a hostPort allows a service to become attached to the hostIP:hostPort, but this (a) requires that users are explicitly maintaining a list of available/consumed host ports, and; (b) limits the number of instances of said pod/pod-template to 1 per-host.

the enhancement here is to somehow allow an end user to specify that their pod/pod-template wants a hostPort, but without explicitly naming that port. mesos supports this by offering up a range of available ports to the k8s-mesos scheduler, but there's no way for the scheduler to know when a user wants a hostPort dynamically assigned from the available port range.

i've previously suggested a magic hostPort of 1 to indicate to the scheduler that dynamic hostPort assignment is desired by the end-user. upon observing a value of 1 the scheduler would dynamically allocate a host port from the available offers and update the pod's desired state to reflect the newly allocated port.

this seems extremely hackish, and limits the real-world use of hostPort 1 by this particular mesos scheduler. however it is backwards compatible with k8s existing use of hostPort 0 as well as with the existing JSON API (no new fields are required).

an alternative approach is to require an end-user to apply a magic label (something like "dynamic-host-port" with, perhaps, a boolean value of true/false) to obtain this behavior. initially this seems cleaner than the magic hostPort=1 approach, but seems to start us down a path of using labels to obtain specific scheduler behavior -- and I'm not sure that's the direction we want to go.

@adam-mesos @ConnorDoyle Thoughts?

### PR DESCRIPTION
**Pull request by jdef**
- resolves #59 
- fixes bugs in proxy deployment script

TODO:
- [x] a few more unit tests, specially around the new code in endpoint_controller

### CODE REVIEW

**Comment by jdef** [comment]
@ConnorDoyle @joerg84 PTAL

**Comment by joerg84** [comment]
Ship It.

File: examples/service.json [hunk 1]
```diff
@@ -3,6 +3,7 @@
   "kind": "Service",
   "apiVersion": "v1beta1",
   "port": 8000,
+  "containerPort": "http",
```
Comment (ConnorDoyle) [inline examples/service.json]: Is this value supposed to be an integer?

Comment (jdef) [inline examples/service.json]: k8s has a nifty IntOrString type. that's what the type of containerPort is:

https://github.com/GoogleCloudPlatform/kubernetes/blob/release-0.14/pkg/api/v1beta1/types.go#L739

Comment (ConnorDoyle) [inline examples/service.json]: Cool, thanks.

**Comment by ConnorDoyle** [comment]
+1, LGTM.  Nice tests!

**Pull request merged**